gen-dirs.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -e
+cd templates
+FILES=$(find -follow)
+LATEST="9.6"
+
+pre_gen_cleanup() {
+    if [[ ${VERSIONS} == *${LATEST}* ]]; then
+        rm -rf ../latest
+    fi
+    for version in ${VERSIONS}; do
+        rm -rf ../${version}
+    done
+}
+
+# postgresql specific
+pg_generate_symlinks() {
+    echo "Creating symlinks:"
+    for version in ${VERSIONS}; do
+        cd ../${version}
+        echo "    ${version}/README.md -> ${version}/root/usr/share/container-scripts/postgresql/README.md"
+        ln -s root/usr/share/container-scripts/postgresql/README.md README.md
+        echo "    ${version}/test test"
+        ln -s ../test test
+    done
+    cd ..
+    if [[ ${VERSIONS} == *${LATEST}* ]]; then
+        echo "Moving ${LATEST} version to latest"
+        if [ -d "./latest" ]; then
+            echo "Failed to copy directory: ./latest already exists"
+            exit 1
+        fi
+        mv ${LATEST} ./latest
+        echo "Creating symlink ${LATEST}->latest"
+        ln -s latest ${LATEST}
+    fi
+}
+
+generate_file() {
+    if [ -L $1 ]; then
+        echo "Skipping symbolic link: $1"
+        return
+    elif [[ -d $1 && ! -d ../${version}/$1 ]]; then
+        mkdir ../${version}/$1
+    else
+        echo "Generating: ${version}/$1"
+        /usr/bin/dg --max-passes 1 --multispec ../specs/multispec.yml \
+        --template $1 --distro centos-7-x86_64.yaml \
+        --multispec-selector version=${version} --output ../${version}/$1
+    fi
+    if [ -x $1 ]; then
+        chmod +x ../${version}/$1
+    fi
+}
+
+generate_dockerfiles() {
+    mkdir ../$1
+    echo "Generating Dockerfile: $1/Dockerfile"
+    /usr/bin/dg --max-passes 25 --template ${PWD}/../Dockerfile.template \
+        --multispec ../specs/multispec.yml --distro centos-7-x86_64.yaml \
+        --multispec-selector version=$1 --output ../$1/Dockerfile
+    echo "Generating Dockerfile: $1/Dockerfile.rhel7"
+    /usr/bin/dg --max-passes 25 --template ${PWD}/../Dockerfile.template \
+        --multispec ../specs/multispec.yml --distro rhel-7-x86_64.yaml \
+        --multispec-selector version=$1 --output ../$1/Dockerfile.rhel7
+
```

Comment:
I'm not sure I follow right, but I think the goal is to remove the generated products from this repo, right?

I would really prefer to not list centos/rhel dockerfiles here, and anywhere else.  We'll have fedora dockerfiles soon, and maybe scls+fedora? Who knows..  So it would be nice if, when generating Dockerfile, the $OS variable was respected.